core/src/main/java/io/kestra/core/queues/QueueService.java

```diff
@@ -16,52 +12,24 @@ public class QueueService {
-        } else if (object.getClass() == WorkerTask.class) {
-            return ((WorkerTask) object).getTaskRun().getId();
-        } else if (object.getClass() == WorkerTaskRunning.class) {
-            return ((WorkerTaskRunning) object).getTaskRun().getId();
         } else if (object.getClass() == WorkerInstance.class) {
             return ((WorkerInstance) object).getWorkerUuid();
-        } else if (object.getClass() == WorkerTaskResult.class) {
-            return ((WorkerTaskResult) object).getTaskRun().getId();
         } else if (object.getClass() == LogEntry.class) {
             return null;
-        } else if (object.getClass() == Flow.class) {
-            return ((Flow) object).uid();
-        } else if (object.getClass() == Template.class) {
-            return ((Template) object).uid();
-        } else if (object instanceof ExecutionKilled) {
-            return ((ExecutionKilled) object).uid();
-        } else if (object.getClass() == Trigger.class) {
-            return ((Trigger) object).uid();
-        } else if (object.getClass() == MultipleConditionWindow.class) {
-            return ((MultipleConditionWindow) object).uid();
         } else if (object.getClass() == SubflowExecution.class) {
             return ((SubflowExecution<?>) object).getExecution().getId();
         } else if (object.getClass() == SubflowExecutionResult.class) {
             return ((SubflowExecutionResult) object).getExecutionId();
-        } else if (object.getClass() == ExecutionDelay.class) {
-            return ((ExecutionDelay) object).uid();
         } else if (object.getClass() == ExecutorState.class) {
             return ((ExecutorState) object).getExecutionId();
         } else if (object.getClass() == Setting.class) {
             return ((Setting) object).getKey();
         } else if (object.getClass() == Executor.class) {
-          return ((Executor) object).getExecution().getId();
-        } else if (object.getClass() == FlowTopology.class) {
-            return ((FlowTopology) object).uid();
+            return ((Executor) object).getExecution().getId();
         } else if (object.getClass() == MetricEntry.class) {
             return null;
-        } else if (object.getClass() == WorkerTrigger.class) {
-            return ((WorkerTrigger) object).getTriggerContext().uid();
-        } else if (object.getClass() == WorkerTriggerRunning.class) {
-            return ((WorkerTriggerRunning) object).getTriggerContext().uid();
-        } else if (object.getClass() == WorkerTriggerResult.class) {
-            return ((WorkerTriggerResult) object).getTriggerContext().uid();
-        } else if (object.getClass() == ExecutionQueued.class) {
-            return ((ExecutionQueued) object).uid();
-        } else if (object.getClass() == ServiceInstance.class) {
-            return ((ServiceInstance) object).id();
+        } else if (object instanceof HasUID hasUID) {
+            return hasUID.uid();
```

Comment:
This is expected to be the case for much of the objects, better to pull this check first (for perf and readability)